enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+  sort of enable, but there seems to be reluctance to expand the cluster profile use case to include
+  these sorts of things.
+  b) enable/disable specific configurations/topologies such as "HA", where components could contribute multiple
+  deployment definitions for different configurations and then the installer/CVO would select the correct
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+Current plan is to constrain this functionality to capability level controls.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because while technically
+[the CVO can delete resources](https://github.com/openshift/enhancements/blob/3db41c0ad1d7960ca66f79a60b962c9b1ec1753c/enhancements/update/object-removal-manifest-annotation.md)
+that are annotated in a particular way, so it could use the same logic to "delete" resources that are now
+matching a filter, just deleting the resources for the component isn't sufficient, as the component also
+needs to clean itself up in case it created any additional resources on the cluster or contributed any
+configuration.
+
+Therefore we plan to support turning a component on, but not turning it off.
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  
+
+There should be no implication here, the CVO has the list of annotations it will filter based on, if the
+new resources match those annotations, the new resources will also be filtered(never applied to the cluster).
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+5. What if a user specifies a capability name to exclude that doesn't match anything?
+
+The installer won't know it's not going to match anything, so it should do nothing.  The CVO
+can tell it doesn't match anything, so it can provide a warning on the clusterversion resource, but
+it should not be an error since a user might put a keyword into the list in anticipation of upgrading
+to a new OCP version where the keyword will match (and thus exclude) a new capability, or perhaps
+the keyword used to match a capability that no longer exists after an upgrade.
```

Comment:
I think we should favor install time correctness over anticipation of future defined capabilities. If we really need to ensure that they cannot upgrade into undesired capabilities we should treat that concern independently by requiring the admin to accept new capabilities or define their intent for any newly defined capabilities.

Reply:
Sorting through this is my only outstanding concern I'd like to see resolved before approving. We have an incredible amount of feedback that the inability to validate install time inputs leads to customer dissatisfaction, starting over due to a typo'd capability seems pretty bad.

Reply:
After discussing on Slack with @staebler @wking @dhellmann it seems like there's consensus that we will have the Installer ensure that listed capabilities are valid. We will advise anyone wishing not to upgrade into additional capabilities to use `inclusionDefault: exclude` and enumerate all desired capabilities.

Reply:
adding this to the EP